lib/node_modules/@stdlib/math/base/special/expm1/src/main.c

```diff
@@ -0,0 +1,215 @@
+}
+
+// END: polyval_q
+
+/* End auto-generated functions. */
+
+/**
+* Computes `exp(x) - 1`.
+*
+* @param x    input value
+* @return	  output value
+*
+* @example
+* double v = stdlib_base_expm1( 0.0 );
+* // returns 0.0
+*/
+double stdlib_base_expm1( const double x ) {
+	double halfX;
+	int sign;
+	double lo;
+	uint32_t hi;
+	uint32_t hx;
+	double r1;
+	double xc = x;
+	double y;
+	double z;
+	double c;
+	double t;
+	double e;
+	int32_t k;
+
+	if ( xc == STDLIB_CONSTANT_FLOAT64_PINF || stdlib_base_is_nan( xc ) ) {
+		return xc;
+	}
+	if ( xc == STDLIB_CONSTANT_FLOAT64_NINF ) {
+		return -1.0;
+	}
+	if ( xc == 0.0 ) {
+		return xc; // handles +-0
+	}
+	// Set y = |x|:
+	if ( xc < 0.0 ) {
+		sign = 1;
+		y = -xc;
+	} else {
+		sign = 0;
+		y = xc;
+	}
+	// Filter out huge and non-finite arguments...
+	if ( y >= LN2x56 ) { // if |x| >= 56*ln(2)
+		if ( sign == 1 ) { // if x <= -56*ln(2)
+			return -1.0;
+		}
+		if ( y >= OVERFLOW_THRESHOLD ) { // if |x| >= 709.78...
+			return STDLIB_CONSTANT_FLOAT64_PINF;
+		}
+	}
+	// Extract the more significant bits from |x|:
+	stdlib_base_float64_get_high_word( y, &hx );
+
+	// Argument reduction...
+	if ( y > STDLIB_CONSTANT_FLOAT64_HALF_LN_TWO ) {
+		if ( y < LN2_HALFX3 ) {
+			if ( sign == 1 ) {
+				hi = xc + LN2_HI;
+				lo = -LN2_LO;
+				k = -1;
+			} else {
+				hi = xc - LN2_HI;
+				lo = LN2_LO;
+				k = 1;
+			}
+		} else {
+			if ( sign == 1 ) {
+				k = ( LN2_INV * xc ) - 0.5;
+			} else {
+				k = ( LN2_INV * xc ) + 0.5;
+			}
+			t = k;
+			hi = xc - ( t * LN2_HI ); // t*ln2_hi is exact here
+			lo = t * LN2_LO;
+		}
+		xc = hi - lo;
+		c = ( hi - xc ) - lo;
+	}
+	// If |x| < 2**-54 => high word: 0 01111001001 00000000000000000000 => 0x3c900000 = 1016070144  => exponent = 01111001001 = 969 = 1023-54
+	else if ( hx < 1016070144 ) {
+		return xc;
+	} else {
+		k = 0;
+	}
+	// x is now in primary range...
+	halfX = 0.5 * xc;
+	z = xc * halfX;
+
+	r1 = 1.0 + ( z * polyval_q( z ) );
+
+	t = 3.0 - ( r1 * halfX );
+	e = z * ( ( r1 - t ) / ( 6.0 - ( xc * t ) ) );
+	if ( k == 0 ) {
+		return xc - ( ( xc * e ) - z ); // c is 0
+	}
+	e = ( xc * ( e - c ) ) - c;
+	e -= z;
+	if ( k == -1 ) {
+		return ( 0.5 * ( xc - e ) ) - 0.5;
+	}
+	if ( k == 1 ) {
+		if ( xc < -0.25 ) {
+			return -2.0 * ( e - ( xc + 0.5 ) );
+		}
+		return 1.0 + ( 2.0 * ( xc - e ) );
+	}
+	if ( k <= -2 || k > 56 ) { // suffice to return exp(x)-1
+		y = 1.0 - ( e - xc );
+
+		// Add k to y's exponent:
+		stdlib_base_float64_get_high_word( y, &hi );
+		hi += ( k << 20 );
+		stdlib_base_float64_set_high_word( hi, &y );
+
+		return y - 1.0;
+	}
+	t = 1.0;
+	if ( k < 20 ) {
+		// 0x3ff00000 - (0x200000>>k) = 1072693248 - (0x200000>>k) => 0x200000 = 0 00000000010 00000000000000000000
+		hi = 1072693248 - ( 0x200000 >> k );
+		stdlib_base_float64_set_high_word( hi, &t ); //t=1-2^-k
+		y = t - ( e - xc );
+	} else {
+		hi = ( ( STDLIB_CONSTANT_FLOAT64_EXPONENT_BIAS - k ) << 20 );
+		stdlib_base_float64_set_high_word( hi, &t ); //t=1-2^-k
```

Comment:
```suggestion
		stdlib_base_float64_set_high_word( hi, &t ); // t=2^-k
```

See JS implementation.